### ISSUE TITLE
Incorrectly deleting imports used exclusively in the `<template>` part of Vue SFCs

### ISSUE THREAD

**Issue by simensol** [simonhaenisch/prettier-plugin-organize-imports#71]
I have 3.1.0 installed, and it seems that enums used in the `<template>` part of Vue SFCs are incorrectly deleted. Here is a minimal working example:

`package.json`:
```json
...
    "devDependencies": {
        "@volar/vue-typescript": "^0.40.1",
        "prettier": "^2.7.1",
        "prettier-plugin-organize-imports": "^3.1.0",
...
```

`types.ts`:
```typescript
export enum ThemeColor {
    PRIMARY = "primary",
    SECONDARY = "secondary",
    ACCENT = "accent",
    ERROR = "error",
    INFO = "info",
    SUCCESS = "success",
    WARNING = "warning"
}
```

`sfc.vue`:
```vue
<script setup lang="ts">
import { ThemeColor } from "@/types"
</script>

<template>
    Color is: {{ ThemeColor.PRIMARY }}
</template>
```

`import { ThemeColor } from "@/types"` is incorrectly deleted. If I add `const color = ThemeColor.PRIMARY` to the `<script setup>` part of `sfc.vue`, the import is not deleted.

EDIT: The same seems to happen with functions used exclusively in the `<template>` part of Vue SFCs:

`helpers.ts`:
```typescript
export function helloWorld(): string {
    return "Hello Worlds"
}
```

`sfc.vue`:
```vue
<script setup lang="ts">
import { helloWorld } from "@/helpers"
</script>

<template>
    {{ helloWorld() }}
</template>
```

Here, `import { helloWorld } from "@/helpers"` is deleted. However, if I add `const hello = helloWorld()` to the `<script setup>` part, the import is not deleted.

**Comment by simonhaenisch** [comment]
Hey can you raise this issue in the [Volar](https://github.com/johnsoncodehk/volar) repo instead, since this is an upstream issue with `@volar/vue-typescript`. In the meantime you can enable the "skip destructive code actions" option (see readme) if you want to disable removing unused imports. Thanks (:

/cc @johnsoncodehk

**Comment by simensol** [comment]
Are you sure Volar is the problem? I use Volar myself, and if I try to import (but not use) a `Test` enum, it seems to know that `ThemeColor` is used in `<template>` and `Test` is not:

```vue
<script setup lang="ts">
import { Test, ThemeColor } from "@/utilities/types"
</script>

<template>Color is: {{ ThemeColor.PRIMARY }}</template>
```

<img width="280" alt="Test_vue_—_frontend__Dev_Container__frontend_" src="https://user-images.githubusercontent.com/4491653/186153964-1fe7d89a-0320-4cde-982f-33c014162fcb.png">

**Comment by simensol** [comment]
Ok, I will open an issue In Volare then (and sorry for the extra issue raised by mistake).

**Comment by simonhaenisch** [comment]
Actually, I'll reopen this, and once the upstream issue is fixed, I'll bump the peer dependency version to the fixed one.

**Comment by simonhaenisch** [comment]
Just published `3.1.1` which has a higher peer dependency version requirement for `@volar/vue-typescript` to fix this. You can just run `npm install -D @volar/vue-typescript@latest` to get the fix.

**Comment by simensol** [comment]
Works like a charm! Thanks!
